apps/emqx_management/test/emqx_mgmt_clients_api_SUITE.erl

```diff
@@ -62,6 +62,29 @@ t_clients(_) ->
     ?assertEqual(ClientsLimit, emqx_mgmt:max_row_limit()),
     ?assertEqual(ClientsCount, 2),
 
+    timer:sleep(300),
+
+    %% get /clients with time(rfc3339)
+    NowTimeStampInt = erlang:system_time(millisecond),
+    %% Do not uri_encode `=` to `%3D`
+    NowTimeString   = emqx_http_lib:uri_encode(binary:bin_to_list(emqx_mgmt_api_clients:unix_ts_to_rfc3339_bin(NowTimeStampInt))),
+    Parameters      = [Param ++ NowTimeString
+                       || Param <- [ "lte_created_at="
+                                   , "lte_connected_at="
+                                   , "gte_created_at="
+                                   , "gte_connected_at="]],
+    RequestResults  = [emqx_mgmt_api_test_util:request_api(get, ClientsPath, Param, AuthHeader)
+                       || Param <- Parameters],
+    DecodedResults  = [emqx_json:decode(Response, [return_maps])
+                       || {ok, Response} <- RequestResults],
+    [ok, ok] = [?assert(
+                   emqx_mgmt_api_clients:rfc3339_to_unix_ts_int(maps:get(<<"created_at">>, hd(maps:get(<<"data">>, LteResponseDecoded)), undefined))
+                   < NowTimeStampInt)
+                || LteResponseDecoded <- element(1, lists:split(2, DecodedResults))],
```

Comment:
A more idiomatic solution is to use pattern matching instead of `hd()`, `lists:split(2, ...)`, `element(1, ...)`, etc.

```suggestion
                || [LteResponseDecoded|_] <- DecodedResults)]
```

Reply:
Also you can use pattern matching to extract the map fields:

```erlang
#{<<"created_at">> := CreatedAt, <<"data">> := [Data|_]} <- DecodedResults
```

...And skip `maps:get`erl